client/utils/auth.py

```diff
@@ -30,6 +28,25 @@
 </html>
 """
 
+def pick_free_port():
+    import socket
+    port_guess = 7777
+    while True:
```

Comment:
I'm a little wary of having no definitive stopping point. Consider aborting after N number of attempts.